### PR TITLE
New Resources: IAMUsersMFA, SSMPatchBaselineGroup and IoT associations and fixes

### PR DESCRIPTION
**Pull request by jakubincloud**
Hi,

We are using aws-nuke for our Cloud vending machine at Cloudreach. This allows us to test aws-nuke with various resources. The result of that are new resources as:

IAMUserMFAattachment  
SSMPatchBaselineGroup
IoTCertificate - ( deactivate Certificate before deletion )
IoTPolicyPrincipalAttachment
IoTThingPrincipalAttachment

### CODE REVIEW

**Comment by jakubincloud** [comment]
The travis has problems with linting in golang 1.11
https://github.com/golang/go/issues/28291

**Comment by jakubincloud** [comment]
The travis has problems with linting in golang 1.11
https://github.com/golang/go/issues/28291

**Comment by jakubincloud** [comment]
Right now Github has issues with the storage layer so some comments / PR are not available.

the PR #285 (update golang from 1.8 to 1.11 )  makes the travis run successful.

File: resources/iam-user-mfa-attachment.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		if err != nil {
+			return nil, err
+		}
+		resources = append(resources, userMfaDevices...)
+	}
+
+	return resources, nil
+}
+
+func (f *IAMUser) ListAttachedMFADevices() ([]Resource, error) {
```
Comment (bjoernhaeuser) [inline resources/iam-user-mfa-attachment.go]: Is there no way to list all devices? This is in the end n+1 thingy here, and we are calling the API for every user once. Could slow down a lot.

Comment (svenwltr) [reply resources/iam-user-mfa-attachment.go]: That's true. I think the `ListMFADevice` just does not need to have a `UserName` argument. At least [the docs](https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#ListMFADevicesInput) do not mention it and the equivalent CLI command `aws iam list-mfa-devices` does not need it.

Comment (jakubincloud) [reply resources/iam-user-mfa-attachment.go]: and I guess the same thing goes for ssh keys ( new thing in IAM that also prevents removing an user completely).
I just saw other comments and I will work on the proposed changes.

**Comment by jakubincloud** [comment]
Please close this PR as I will submit new changes in a separate PRs. Thanks

**Pull request closed without merging**